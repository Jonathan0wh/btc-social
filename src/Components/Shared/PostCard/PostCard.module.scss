.title {
  font-size: large;
  font-weight: bold;
}

.link,
.link:hover {
  color: white;
  text-decoration: none;
}
