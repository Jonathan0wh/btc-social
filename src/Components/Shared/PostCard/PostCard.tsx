import React from 'react';
import { Link } from 'react-router-dom';
import {
  Card,
  CardTitle,
  CardHeader,
  CardBody,
  CardText,
  CardFooter,
  Button
} from 'reactstrap';

import styles from './PostCard.module.scss';

interface PostCardProps {
  id: number;
  title: string;
  body: string;
  userId: number;
  showCommentButton?: boolean;
}

const PostCard = ({
  id,
  title,
  body,
  userId,
  showCommentButton = false
}: PostCardProps) => {
  return (
    <Card>
      <CardHeader>
        <CardTitle className={styles.title}>{title}</CardTitle>
      </CardHeader>
      <CardBody>
        <CardText>{body}</CardText>
        {showCommentButton && (
          <Button>
            <Link to={`post/${id}`} className={styles.link}>
              Comment
            </Link>
          </Button>
        )}
      </CardBody>
      <CardFooter>Posted by {userId}</CardFooter>
    </Card>
  );
};

export default PostCard;
