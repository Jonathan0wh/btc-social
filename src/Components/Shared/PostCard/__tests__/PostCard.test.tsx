import React from 'react';
import { shallow } from 'enzyme';
import PostCard from '../PostCard';

describe('<PostCard />', () => {
  it('should render without comment button', () => {
    const wrapper = shallow(<PostCard title="test" body="" userId={0} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with comment button', () => {
    const wrapper = shallow(
      <PostCard title="test" body="" userId={0} showCommentButton />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
