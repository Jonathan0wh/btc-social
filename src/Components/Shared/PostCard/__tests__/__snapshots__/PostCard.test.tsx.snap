// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should render with comment button 1`] = `
<Card
  tag="div"
>
  <CardHeader
    tag="div"
  >
    <CardTitle
      className="title"
      tag="div"
    >
      test
    </CardTitle>
  </CardHeader>
  <CardBody
    tag="div"
  >
    <CardText
      tag="p"
    />
    <Button
      color="secondary"
      tag="button"
    >
      <Link
        className="link"
        to="post/undefined"
      >
        View Comments
      </Link>
    </Button>
  </CardBody>
  <CardFooter
    tag="div"
  >
    Posted by 
    0
  </CardFooter>
</Card>
`;

exports[`<PostCard /> should render without comment button 1`] = `
<Card
  tag="div"
>
  <CardHeader
    tag="div"
  >
    <CardTitle
      className="title"
      tag="div"
    >
      test
    </CardTitle>
  </CardHeader>
  <CardBody
    tag="div"
  >
    <CardText
      tag="p"
    />
  </CardBody>
  <CardFooter
    tag="div"
  >
    Posted by 
    0
  </CardFooter>
</Card>
`;
